src/Compilers/Core/Portable/Compilation/SemanticModel.cs

```diff
@@ -876,5 +876,15 @@ protected internal virtual SyntaxNode GetTopmostNodeForDiagnosticAnalysis(ISymbo
         {
             return declaringSyntax;
         }
+
+        /// <summary>
+        /// Root of this semantic model
+        /// </summary>
+        internal SyntaxNode Root => RootCore;
+
+        /// <summary>
+        /// Root of this semantic model
+        /// </summary>
+        protected abstract SyntaxNode RootCore { get; }
```

Comment:
I followed pattern Semantic model already has. for example, GetSymbolInfo is internal, but GetSymbolInfoCore is protected, and there are many more code with same pattern.

http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/SemanticModel.cs,113

since SemanticModel contains internal abstract member, user can't create its own type inheriting SemanticModel, so protected RootCore being inside public api won't matter much?